controllers/storagecluster/initialization_reconciler_test.go

```diff
@@ -236,13 +342,24 @@ func createFakeInitializationStorageClusterReconcilerWithPlatform(t *testing.T,
 		},
 	}
 
+	statusSubresourceObjs := []client.Object{sc}
+	if storageProfiles != nil {
+		for _, sp := range storageProfiles.Items {
+			statusSubresourceObjs = append(statusSubresourceObjs, &sp)
+		}
+	}
+
 	obj = append(obj, mockNodeList.DeepCopy(), cbp, cfs, cnfs, cnfsbp, cnfssvc, infrastructure, networkConfig)
-	client := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(obj...).WithStatusSubresource(sc).Build()
+	client := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(obj...).WithStatusSubresource(statusSubresourceObjs...).Build()
```

Comment:
I'm wondering whether I've stumped upon a bug here: the only way I was able to get the `StorageProfile`s' status to go through the update via the `StorageCluster`'s reconciler [below](https://github.com/red-hat-storage/ocs-operator/blob/fusion-hci-4.14/controllers/storagecluster/cephfilesystem.go#L71-L85) was to append them along with `StorageCluster` here. Usually it's a one-to-one ratio for resources that have their own controller, e.g. for `OCSInit` that's also being `watch`ed by `StorageCluster`. 
```go
r.Client.Status().Update(r.ctx, &storageProfile)
``` 
@nb-ohad thoughts?